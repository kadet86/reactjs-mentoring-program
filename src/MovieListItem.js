import React from 'react';

export default function MovieListItem({movie, navigateToMovie}) {
    return (
        <div className="movie-list-item">
            <a onClick={() => navigateToMovie(movie)}>
                <img src={movie.poster_path} />
            </a>
            <div>
                <h2>{movie.title}</h2>
                <span>{movie.release_date.substr(0, 4)}</span>
            </div>
            <div>{movie.genres.join(' & ')}</div>
        </div>
    );
}