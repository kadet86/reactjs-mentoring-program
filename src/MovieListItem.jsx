import Link from 'next/link';
import React from 'react';

export default function MovieListItem({ movie, query }) {
    return (
        <div className="movie-list-item">
            <Link href={`/film?id=${movie.id}`} as={`/film/${movie.id}`}>
                <img src={movie.poster_path} />
            </Link>
            <div>
                <h2>{movie.title}</h2>
                <span className="movie-list-item__date">
                    {movie.release_date && movie.release_date.substr(0, 4)}
                </span>
            </div>
            <div className="movie-list-item__genres">
                {movie.genres && movie.genres.join(' & ')}
            </div>
        </div>
    );
}
