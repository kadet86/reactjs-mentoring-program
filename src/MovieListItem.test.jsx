import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { shallow } from 'enzyme';
import MovieListItem from './MovieListItem';

describe('MovieListItem component', () => {
    it('has persistent snapshot', () => {
        const movie = {
            id: 1,
            release_date: '2018-02-03',
            genres: ['action', 'comedy'],
        };

        const component = shallow(
            <MemoryRouter>
                <MovieListItem movie={movie} />
            </MemoryRouter>
        );

        expect(component.html()).toMatchSnapshot();
    });
});
