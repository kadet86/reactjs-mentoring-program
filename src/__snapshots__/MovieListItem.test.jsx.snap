// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieListItem component has persistent snapshot with movie={{}} 1`] = `"<div class=\\"movie-list-item\\"><a><img/></a><div><h2></h2><span class=\\"movie-list-item__date\\"></span></div><div class=\\"movie-list-item__genres\\"></div></div>"`;

exports[`MovieListItem component has persistent snapshot with movie={{genres:["action", "comedy"]}} 1`] = `"<div class=\\"movie-list-item\\"><a><img/></a><div><h2></h2><span class=\\"movie-list-item__date\\"></span></div><div class=\\"movie-list-item__genres\\">action &amp; comedy</div></div>"`;

exports[`MovieListItem component has persistent snapshot with movie={{release_date:"2018-02-03"}} 1`] = `"<div class=\\"movie-list-item\\"><a><img/></a><div><h2></h2><span class=\\"movie-list-item__date\\">2018</span></div><div class=\\"movie-list-item__genres\\"></div></div>"`;
